dipy/reconst/dki.py

```diff
@@ -1676,8 +1680,14 @@ def fit(self, data, mask=None):
             return self.multi_fit(data_thres, mask=mask)
 
         params, extra = self.fit_method(self.design_matrix, data_thres,
-                                 *self.args, **self.kwargs)
-        return DiffusionKurtosisFit(self, params)
+                                        return_S0_hat=self.return_S0_hat,
+                                        *self.args, **self.kwargs)
+
+        S0_params = None
+        if self.return_S0_hat:
+            params, S0_params = params
+
+        return DiffusionKurtosisFit(self, params, S0_params)
```

Comment:
Can you do `model_S0=S0_params`?  

In april, we will force the codebase to be keyword-only friendly. So this is just to anticipate that.

Reply:
is this because `model_S0` is a keyword param, and we want to make that clear here?

Reply:
yes, and we will do it for the whole codebase and examples starting in April.

We plan to follow https://peps.python.org/pep-3102/